File: C++/21_insertionSort.cpp
#include<iostream>
using namespace std;

void InsertionSort(int* arr, int n){
    for(int i = 1; i<n; i++){
        int key = *(arr+i);
        int j = i-1;

        while (j>=0 && *(arr+j) > key){
            *(arr+j+1) = *(arr+j);
            j--;
        }
        *(arr+j+1) = key;

    }
}  

int main(void){
    int n,i;
    cout<<"Enter array size: ";
    cin >> n;
    int arr[n];
    cout<<"Enter array element: \n";
    for( i = 0; i < n; i++){
        cout<<"Enter element "<<i<<" : ";
        cin>>arr[i];
    }
    cout<<"Array before sorting: ";
    for( i = 0; i < n-1; i++){
        cout<<arr[i]<<", ";
    }
    cout<<arr[i]<<endl;
    InsertionSort(arr, n);
    cout<<"Insertion Sort done\n";
    cout<<"Array after sorting: ";
    for(i = 0; i < n-1; i++){
        cout<<arr[i]<<", ";
    }
    cout<<arr[i]<<endl;
    return 0;
}

File: Python/21_insertionSort.py

def insertionSort(arr, n):
    for i in range(n-1):
        key = arr[i]
        j = i-1
        while (j >= 0 and arr[j] > key):
            arr[j+1] = arr[j]
            j -= 1
        arr[j+1] = key


if __name__ == '__main__':
    n = int(input("Enter the number of elements in array:  "))
    a = []

    for i in range(n):
        a.append(int(input(f"Enter element {i}: ")))
    # a = [2,1,4,3]
    # n = len(a)
    print("Array before Sorting: ", a)
    insertionSort(a, n)
    print("Array after Sorting: ", a)
